### PR TITLE
Core-cardmodule fix

### PR DESCRIPTION
**Pull request by kds1660**
1. Fixed in cardmodule error in setNewBalance method
2. Add in core setNewBalance after cash retrieving
3. Change - now working card date check and add in core

### CODE REVIEW

File: js/controller/core.js [hunk 1]
```diff
@@ -200,6 +202,7 @@ function Core(cashModule, cardModule, navigation,cashOutModule) {
                         var cashOut = cashModule.getCash(+cash);
                         navigation.showMessage(STATE_TEXT.ALL_OK);
                         cashOutModule.showMoney(cashOut);
+                        cardModule.setNewBalance(+cash);
```
Comment (Varkh) [inline js/controller/core.js]: I would prefer `parseInt` function

File: js/model/cardDataModel.js [hunk 2]
```diff
@@ -67,7 +67,7 @@ function CardDataModel(holderName, cardNumber, expirationDate, ballance) {
      * Metod: sets new ballance
      */
     this.setBalance = function (newBallance) {
-        this.ballance = newBallance;
+        this._ballance = newBallance;
```
Comment (Varkh) [inline js/model/cardDataModel.js:72]: I think it's better to have `balance` as private field(not in this)

File: js/model/cardDataModel.js [hunk 3]
```diff
@@ -87,13 +87,17 @@ function CardDataModel(holderName, cardNumber, expirationDate, ballance) {
 
 CardDataModel.prototype.isNotExpired = function () {
     var now = new Date();
```
Comment (Varkh) [inline js/model/cardDataModel.js:91]: Create `expirationDate` with `new Date`
and just check
```
now.getTime() < expirationDate.getTime()
```

**Review by Varkh — CHANGES_REQUESTED** [review]

File: js/model/cardDataModel.js (hunk 2 above)
Comment (kds1660) [reply js/model/cardDataModel.js:72]: first we must implement traffic card in atm, for now  - we creating new card for every step, i guess - in main.js create 3 objects-cards and working with it - push in atm and core must back this objects ande change event card inserted - in details object, in core add event card-ejected and details - card object. something like this

**Comment by Varkh** [comment]
Tests are failing

File: css/main.css [hunk 4]
```diff
@@ -131,4 +131,20 @@ p {
 .addedInput{
     width: 110px;
     text-align: center;
-}
\ No newline at end of file
+}
+
+table {
```
Comment (Varkh) [inline css/main.css:136]: don't use styles for elements, use classes
what if one more table will be added

**Comment by kds1660** [comment]
var event = new CustomEvent не прошел, пока закомменил


2016-12-07 22:08 GMT+02:00 Denys <notifications@github.com>:

File: js/model/cardDataModel.js [hunk 5]
```diff
@@ -60,14 +62,14 @@ function CardDataModel(holderName, cardNumber, expirationDate, ballance) {
     };
 
     this.getBallance = function () {
-        return ballance;
+        return self._ballance;
```
Comment (Varkh) [inline js/model/cardDataModel.js:65]: not good.
Was better before

File: js/model/cardDataModel.js (hunk 2 above)
Comment (Varkh) [reply js/model/cardDataModel.js:72]: Leave all card logic in card module. Only card module should handle CardModel

File: js/model/cardDataModel.js [hunk 6]
```diff
@@ -87,13 +89,10 @@ function CardDataModel(holderName, cardNumber, expirationDate, ballance) {
 
 CardDataModel.prototype.isNotExpired = function () {
     var now = new Date();
-    if(now.getFullYear() <= this._expirationDate[2]
-        && now.getMonth() < this._expirationDate[1]
-        && now.getDate() < this._expirationDate[0]) {
+    var expDate = new Date (this._expirationDate[2],this._expirationDate[1]-1,this._expirationDate[0])
+    if(now.getTime() < expDate.getTime()) {
```
Comment (Varkh) [inline js/model/cardDataModel.js:93]: Looks better now :)

**Review by Varkh — CHANGES_REQUESTED** [review]

**Review by Varkh — APPROVED** [review]
Approved but changes required

File: js/model/cardDataModel.js [hunk 7]
```diff
@@ -8,6 +8,7 @@
  * @constructor
 **/
 function CardDataModel(holderName, cardNumber, expirationDate, ballance) {
+    var self=this;
```
Comment (Varkh) [inline js/model/cardDataModel.js:11]: remove self

File: js/model/cardDataModel.js [hunk 8]
```diff
@@ -60,14 +62,14 @@ function CardDataModel(holderName, cardNumber, expirationDate, ballance) {
     };
 
     this.getBallance = function () {
-        return ballance;
```
Comment (Varkh) [inline js/model/cardDataModel.js:63]: I want to see:

`var ballance;`
not
`this._ballance;`

File: js/model/cardDataModel.js (hunk 5 above)
Comment (kds1660) [reply js/model/cardDataModel.js:65]: i dont understend for now - this - div with value 500, not find where is this happens, for now change to self

**Review by kds1660 — COMMENTED** [review]
i remember, for first I implement card move in atm, now try remove this._balanse and this remedy situation with self in cardmodel, i guess

**Pull request merged**